deployments/container/Dockerfile.devel

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+ARG GOLANG_VERSION=x.x.x
+FROM golang:${GOLANG_VERSION}
+
+ARG CLIENT_GEN_VERSION=v0.26.1
+ARG CONTROLLER_GEN_VERSION=v0.9.2
+ARG GOLANGCI_LINT_VERSION=v1.52.0
+ARG MOQ_VERSION=v0.3.4
+
+RUN go install sigs.k8s.io/controller-tools/cmd/controller-gen@${CONTROLLER_GEN_VERSION} \
+    && go install k8s.io/code-generator/cmd/client-gen@${CLIENT_GEN_VERSION}
```

Comment:
Do we want all 4 ARGS to be used in the RUN command as well?

Reply:
thanks for catching that, removed them. MOQ and GOLANGCI-LINT are not needed for `make generate` or any other make targets required by the demo

Reply:
I meant the opposite -- you should INCLUDE them here, not remove them. We need at least golangci-lint (as we have a make target for it already), and I can imagine we will want to add moq at some point.

Reply:
included